src/topology/algebra/ordered.lean

```diff
@@ -451,6 +455,178 @@ instance orderable_topology.regular_space : regular_space α :=
       by rw [←sup_principal, inf_sup_left, ht₁a, ht₂a, bot_sup_eq]⟩,
   ..orderable_topology.t2_space }
 
+/-- A set is a neighborhood of `a` if and only if it contains an open interval containing `a`,
+provided `a` is neither a bottom element nor a top element. -/
```

Comment:
In this lemma and others, the docstring doesn't really match the statement. Do you have examples showing the statement is really the convenient one?

Reply:
I have tried to add some details to the docstrings to make them more accurate. My goal is really to have statements that work on the real numbers, as I guess this will be the use case 99% of the time, while still being true in natural numbers or ennreals or whatever. For my use case (proving that the function mapping `x` to `0` if `x ≤ 0` and `exp(-1/x)` if `x > 0` is smooth), these are exactly the statements I need.